cookbooks/aws-parallelcluster-environment/recipes/default.rb

```diff
@@ -1 +1,2 @@
 # Required for ChefSpec tests to be able to converge
+# default['cluster']['scheduler_slots'] = 'vcpus'
```

Comment:
q: why is this commented out? and why is in the default recipe? we should move it to the attributes file of the aws-parallelcluster-environment cookbook

Reply:
As far as I can see, we don't need this value defined as an attribute, since it's used only in scope of `cfnconfig_mixed` recipe.

Reply:
It's better to keep it as attribute, this something related to multithreading and in past it was used by our users, by setting for example.
```
ExtraChefAttributes: '{"cluster": {"scheduler_slots": "cores"}}'
```

Reply:
I kept that as part of trying to test below part of the code in spec file. Was planning on removing it if I found a solution.

```
node.default['cluster']['instance_slots'] = case node['cluster']['scheduler_slots']
                                            when 'vcpus'
                                              node['cpu']['total']
                                            when 'cores'
                                              node['cpu']['cores']
                                            else
                                              node['cluster']['scheduler_slots']
                                            end
```

Reply:
I kept that as part of trying to test this part of the code in spec file. Was planning on removing it if I found a solution.

Reply:
As discussed, will make changes for moving the recipes only.